### ISSUE TITLE
Redirection Error - Still a problem, nginx, multisite, instructions followed

### ISSUE THREAD

**Issue by dandick** [johngodley/redirection#1766]
Redirection Error - Still a problem, nginx, multisite, instructions followed
Cannot update database for redirection.  Turned off all-in-one security and firewall.

The whole https thing makes sense.  There's a front end nginx as a reverse proxy.  Then on the same machine is docker and docker-compose.  This site lives in a docker-compose.yml setup.  It's running SSL.  It's running PHP 7 in a PHP-FPM  setup in a Docker container for the inner nginx/wordpress setup.  There's mariadb which works fine.  There's another container running phpmyadmin and it works fine.  The website works ok for the most part.  It's just I'm getting notifications I need to upgrade the data for Redirection and it won't let me.

I did not see much of anything for https in wp-config.php.  
```
Plugin: 4.2.3
WordPress: 5.2 (multi)
PHP: 7.3.2
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.131 Safari/537.36
JavaScript: http://godsblogs.com/wp-content/plugins/redirection/redirection.js
REST API: http://godsblogs.com/wp-json/

Error: Failed to fetch
```

**Comment by johngodley** [comment]
The https setting is set by the site URL on the Settings > General page. However, you are using multisite and I think that lives somewhere in the network admin - you will be able to change the site URL from http to https.

Alternatively you can change the REST API option in the dropdown inside the error. This may help work around the misconfigured URLs.

**Comment by dandick** [comment]
Thanks, John,

Yes.  It actually gives three options, two of which are red from errors 
and one which is green.  But when I switch to the green one, it doesn't 
change, and it's probably something wacky I've done perhaps with 
permissions or something else--not sure.

That's for the REST API part.  But as for the first thing, I sort of 
cruised through some of the php files--not to change them but to see 
what things looked for or set _SERVER['HTTPS'] or something like that.  
For most of my IT life, I have learned not to toy with the base 
software.  Even if you fix something for yourself and make it work 
better, it could cause other problems or easily be overwritten with the 
next update.  For that reason, I like to go vanilla with any software I 
use even if I could dive in and change the internals and make it work.

But there's always a possibility of a bug or that I messed something up 
in configuration.  But I also have to confess that my installation is a 
bit complex in that it is in docker-compose. That is, there are three 
containers and I have another docker-compose.yml file to fire things up 
with a phpmyadmin as well, but I usually keep that container down.  
Everything else is identical--same containers are started.  But people 
don't connect to the container with nginx directly.  They come in 
through a front end nginx installation and then it proxies connections 
sending them over to the appropriate internal port to reach the nginx 
container.  So you have 2 nginxes that have to deal with the http/https 
thing, or actually redirect browsers to https when an http comes in, 
which I have set up on the front end nginx.  Then the https connection 
is proxied over to the container based nginx installation.  But 
everything gets through to wordpress just fine.  But then wordpress 
multi-site needs to deal with the http/https thing in its own way.  And 
there's the nginx helper plugin.  There are domains that are set up so 
not everything is a subdomain of godsblogs.com.  The djd.godsblogs.com 
is set up with danieljdick.com as it's main domain and djd.godsblogs.com 
simply does not work basically.  The same is true for nodivorces.com. 
That is, there is an ndc subdomain that is set up for nodivorces.com.  
And the rest is all subdomained.  Things are set up properly with google 
domains which at one time were on godaddy.com until I decided to move 
over to google.

Most things seem to work.  It's just that it won't let me update the 
database for the redirect thing.  I ran the mysql_upgrade thing and made 
sure all the tables in the database were "repaired" and "analyzed" just 
for good measure.  However not all tables are of the format I would have 
expected.  But I am not getting the impression that's the problem.

Anyway, I don't want to burden you or anyone with this.  I'm mostly 
sharing because when the database update fails, it allows for people to 
communicate the problem they're experiencing.  So I did not want anyone 
to think it was simply because I did not apply the instructions already 
on hand unless I missed something else.

This is not something I'd consider an emergency.  I'm just picky and 
like everything to be clean, so it sort of challenges me to hunt down 
any warning messages or anything that might cause problems.

Thanks again,

Dan

**Comment by johngodley** [comment]
You could pick the 'manual upgrade' option and follow the instructions and perform the upgrade yourself

**Comment by dandick** [comment]
Thanks, John, I tried that as well. The query it produces and tells you 
to run doesn't seem to have the numbed tables like these.

UPDATE `gbc_redirection_items` SET match_url='regex' WHERE regex=1;
UPDATE `gbc_redirection_items` SET match_url=SUBSTRING_INDEX(LOWER(url), 
'?', 1) WHERE regex=0;
UPDATE `gbc_redirection_items` SET 
match_url=LEFT(match_url,LENGTH(match_url)-1) WHERE regex=0 AND 
match_url != '/' AND RIGHT(match_url, 1) = '/';
UPDATE `gbc_redirection_items` SET match_url='/' WHERE match_url='';
UPDATE `gbc_redirection_items` SET 
match_url=LOWER(LEFT(SUBSTRING_INDEX(url, '?', 
1),LENGTH(SUBSTRING_INDEX(url, '?', 1)) - 1)) WHERE 
RIGHT(SUBSTRING_INDEX(url, '?', 1), 2) = '//' AND regex=0;
UPDATE `gbc_redirection_items` SET match_url='/' WHERE match_url='';

I pressed the "Complete Upgrade" button after running this manually.  It 
still didn't make any change.

**Comment by dandick** [comment]
There were other SQL clippings for other websites where the tables would start with gbc_9_redirection... or some other number of course.

I ended up just deleting all the tables and started over, but I ended up face to face with the http/https issue again as I am using a reverse proxy nginx to a set of three docker containers--one for an inner nginx, a mariadb, and wordpress that I set up multi-site with domains.  DNS is set up properly with Google Domains as it has been for years earlier with Godaddy.  But the subdomains are properly set up.

And in general things work, but just not the redirect.  I got rid of plugins and some themes I thought might be a problem.  I think I'm having a problem simply because there's no simple way to flip a multi-site over to http as far as I can tell and wordpress has to recognize the url with the https perhaps on port 80 with the encryption happening on the reverse proxy out in front, or else I'll need to do a straight pass through and let the backend nginx do the SSL encryption/decryption.  But either way, wordpress needs to know it's dealing in https or putting out local GET and POST requests or links with https, wherever the actual encrypting/decrypting web server may be.

I'm also having some hiccups with the images uploading them and having them show up in the gallery for me to put into my articles.

I know I must be doing something goofy.  Anyway, again, I don't want to bother anyone.  I mostly just want to think out loud while working on this in case it may help someone else or in case someone may think of something that might be obvious that I'm missing.

Thanks again,
Dan

**Comment by dandick** [comment]
Voila!!!  I finally got it working.

There is a "Really Simple SSL" plugin available.  I downloaded it, installed it, and it made all the difference in the world.  I got rid of the plugin that removed the http and https from links as I thought that would solve the duplicate website SEO problem and fix the problem I was experiencing but it didn't.

I'll see how it works for a few days whether any glitches show up or not, but it seems to be working great, so I wanted to share that in case it might help someone else.

Thanks again for your tips, though as they helped me understand a little better what was happening internally.

Dan
